scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,263 @@
+      return getProto(
+        textProtoFileName = "file_content_validation_checks.pb",
+        proto = FileContentChecks.getDefaultInstance()
+      ).getFileContentChecksList()
+    }
+
+    /**
+     * helper function to parse the textproto file to a proto class.
+     *
+     * @param textProtoFileName name of the textproto file to be parsed
+     * @param proto instance of the proto class
+     * @return proto class from the parsed textproto file
+     */
+    private fun <T : MessageLite> getProto(textProtoFileName: String, proto: T): T {
+      val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+      val builder = proto.newBuilderForType()
+
+      @Suppress("UNCHECKED_CAST")
+      val protoObj: T =
+        FileInputStream(protoBinaryFile).use {
+          builder.mergeFrom(it)
+        }.build() as T
+      return protoObj
+    }
+
+    /**
+     * Checks for a prohibited file naming pattern.
+     *
+     * @param repoPath the path of the repo
+     * @param searchFiles a list of all the files which needs to be checked
+     * @param prohibitedFilenameObj proto object of FilenameCheck
+     * @return check failed or passed
+     */
+    private fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: List<File>,
+      prohibitedFilenameObj: FilenameCheck,
+    ): Boolean {
+      val prohibitedFilenameRegex = prohibitedFilenameObj.getProhibitedFilenameRegex().toRegex()
+      val matchedFiles = searchFiles.filter { file ->
+        return@filter file.name !in prohibitedFilenameObj.getExemptedFileNameList()
+          && prohibitedFilenameRegex.matches(
+          RepoFile.retrieveFilePath(
+            file,
+            repoPath)
+        )
+      }
+
+      logProhibitedFilenameFailure(
+        repoPath = repoPath,
+        matchedFiles = matchedFiles,
+        errorToShow = prohibitedFilenameObj.getFailureMessage()
+      )
+
+      return matchedFiles.toList().size != 0
+    }
+
+    /**
+     * Checks for a prohibited file content.
+     *
+     * @param repoPath the path of the repo
+     * @param searchFiles a list of all the files which needs to be checked
+     * @param prohibitedContentObj proto object of FileContentCheck
+     * @return check failed or passed
+     */
+    private fun checkProhibitedContent(
+      repoPath: String,
+      searchFiles: List<File>,
+      prohibitedContentObj: FileContentCheck,
+    ): Boolean {
+      var contentCheckFailedFlag = false
+      val fileNameRegex = prohibitedContentObj.getFilenameRegex().toRegex()
+      val prohibitedContentRegex = prohibitedContentObj.getProhibitedContentRegex().toRegex()
+      val matchedFiles = searchFiles.filter {
+        it.name !in prohibitedContentObj.getExemptedFileNameList()
+          && fileNameRegex.matches(it.name)
+          && File(it.toString()).bufferedReader().lineSequence()
+          .foldIndexed(initial = false) { lineNumber, acc, lineStr ->
```

Comment:
better naming for `acc` and `lineStr` required

Reply:
changed the names, PTAL.